tools/umi_tools/main.nf

```diff
@@ -5,32 +5,38 @@ nextflow.enable.dsl=2
 
 // Process definition
 process umitools_dedup {
-    publishDir "${params.outdir}/umitools/dedup",
-        mode: "copy", overwrite: true
+    publishDir "${params.outdir}/${opts.publish_dir}",
+        mode: "copy", 
+        overwrite: true,
+        saveAs: { filename ->
+                      if (opts.publish_results == "none") null
+                      else filename }
 
     container 'luslab/nf-modules-umi_tools:latest'
 
     input:
-        tuple val(sample_id), path(bam)
+        val(opts)
+        tuple val(meta), path(bam)
        
     output:
-        tuple val(sample_id), path("${sample_id}.dedup.bam"), emit: dedupBam
-        tuple val(sample_id), path("${sample_id}.dedup.bam.bai"), emit: dedupBai
```

Comment:
bam/bai files need to be outputed as a tuple to stay inline with the nf-core standards